modules/Bio/EnsEMBL/Hive/DBSQL/NakedTableAdaptor.pm

```diff
@@ -47,7 +47,18 @@ use base ('Bio::EnsEMBL::Hive::DBSQL::BaseAdaptor');
 sub slicer {    # take a slice of the hashref (if only we could inline in Perl!)
     my ($self, $hashref, $fields) = @_;
 
-    return [ @$hashref{@$fields} ];
+    my $overflow_limit = $self->overflow_limit();
+
+    return [ map { eval { my $value = $hashref->{$_};
```

Comment:
I know there is an `eval` in ObjectAdaptor too, but do we really need them ? I am concerned they could hide some errors